.github/workflows/_lints.yml

```diff
@@ -0,0 +1,77 @@
+name: Reusable workflow for lints
+
+on:
+  workflow_call:
+    inputs:
+      target:
+        required: true
+        type: string
+
+jobs:
+  clippy:
+    name: Clippy (MSRV)
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ${{ inputs.target }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install build dependencies
+        run: sudo apt install libudev-dev
+      - name: Run Clippy
+        uses: auguwu/clippy-action@1.1.2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          working-directory: ${{ inputs.target }}
+          args: --all-features --all-targets -- -D warnings
```

Comment:
This check is specifically against MSRV as pinned in `rust-toolchain.toml`. So the only time new warnings would appear is if your PR is changing that pinned toolchain, and then it's not unrelated to your PR. Once we get onto stable Rust, this will only occur in PRs that bump MSRV.